core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -569,4 +572,41 @@ default SplitHttpResponse split(EventExecutor executor) {
                                         Function<? super HttpData, ? extends ByteBuf> byteBufConverter) {
         return new DecodedHttpStreamMessage<>(this, decoder, alloc, byteBufConverter);
     }
+
+    /**
+     * Transforms the {@link ResponseHeaders} emitted by {@link HttpResponse} by applying the specified
+     * {@link Function}.
+     */
+    default HttpResponse mapHeaders(Function<? super ResponseHeaders, ? extends ResponseHeaders> function) {
+        requireNonNull(function, "function");
+        final StreamMessage<HttpObject> stream =
+                map(obj -> obj instanceof ResponseHeaders ? function.apply((ResponseHeaders) obj) : obj);
```

Comment:
How about applying the `function` only when the `ResponseHeaders` is not informational?
I think there's no chances that a user want to manipulate the information headers.
Also users do not have to check if it's informational in the `function`.

Reply:
I am not sure 100% that it is a good decision to exclude an informational headers from `mapHeaders()` without additional options.
How about adding `mapInformational(Function)` for an informational headers?

Reply:
That also sounds good. 👍